### DISCUSSION TITLE
System identification capabilities

### DISCUSSION THREAD

**Discussion by billtubbs** [python-control/python-control#606] [Ideas]
Not sure if this discussion has been had, but I've been looking for a good system identification package in Python for a while.  As far as I can tell Python-Control has not implemented any of the Matlab functions like `procest`, `ar`, `arx`, `armax`, `idss`, `polyest`, `n4sid`, `pem`, etc.

I don't know any open source software that can compete with Mathwork's System Identification Toolbox so this would be a great thing i.m.o. **_Is there a plan for system identification capabilities in Python-control?_** 

If so, would the goal be to create additional capabilities (e.g. similar to Matlab) or to integrate with these existing projects?  (I'd be interested in contributing to any such effort.)

Sorry, if I have missed something.

**Comment by billtubbs** [comment]
From my brief research there are a few existing system identification packages in Python:

- [GEKKO Optimization Suite](https://gekko.readthedocs.io/en/latest/) – constrained linear/non-linear optimization with API access to a suite of different solvers
- [SIPPY](https://github.com/CPCLAB-UNIPI/SIPPY) – linear models from input-output data
- [SysIdentPy](https://sysidentpy.org) – system Identification using NARMAX models
- [Opty](https://opty.readthedocs.io/en/latest/) – constrained optimization with symbolically-defined ordinary differential equations

Does anyone know of others I should add to this list?

**Comment by murrayrm** [comment]
@jgoppert has a [sysid package](https://github.com/jgoppert/sysid)
